.github/workflows/main.yaml

```diff
@@ -12,7 +12,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest]
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10", "3.12"]
```

Comment:
One last thing, why no 3.11 here?

Reply:
Just because it wasn't there before and I didn't want to create additional testing load, when my focus was 3.12 and not 3.11 (which was already supported).

Though on the other side [I've already found an issue raising on `linkml`](https://github.com/linkml/linkml/pull/1704#issuecomment-1952810752) because `linkml-runtime` wasn't being tested on 3.11. I will make sure to add the test for 3.11 on this [follow-up PR](https://github.com/linkml/linkml-runtime/pull/299). I will add it also once we have an agreement on how to resolve [this conversation](https://github.com/linkml/linkml-runtime/pull/286#pullrequestreview-1894203188).

Reply:
I'm all for incremental, so I'll resolve this conversation, but feel free to make a new PR when this is done (I don't like extra testing load but linkml-runtime is quite light compared to linkml)

Reply:
Ig if we are testing on either side of it, should be fine???